File: generated-code/codecraft/kotlin/src/main/kotlin/model/Vec2Int.kt
package model

import util.StreamUtil

class Vec2Int {
    var x: Int = 0
    var y: Int = 0

    constructor(x: Int, y: Int) {
        this.x = x
        this.y = y
    }

    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, x)
        StreamUtil.writeInt(stream, y)
    }

    override fun toString(): String {
        var stringBuilder = StringBuilder("Vec2Int { ")
        stringBuilder.append("x: ")
        stringBuilder.append(x)
        stringBuilder.append(", ")
        stringBuilder.append("y: ")
        stringBuilder.append(y)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Vec2Int {
            var x: Int
            x = StreamUtil.readInt(stream)
            var y: Int
            y = StreamUtil.readInt(stream)
            return Vec2Int(x, y)
        }
    }
}

File: generated-code/codecraft/cpp/model/Vec2Int.cpp
#include "Vec2Int.hpp"

Vec2Int::Vec2Int() { }

Vec2Int::Vec2Int(int x, int y) : x(x), y(y) { }

Vec2Int Vec2Int::readFrom(InputStream& stream) {
    int x;
    x = stream.readInt();
    int y;
    y = stream.readInt();
    return Vec2Int(x, y);
}

void Vec2Int::writeTo(OutputStream& stream) const {
    stream.write(x);
    stream.write(y);
}

std::string Vec2Int::toString() const {
    std::stringstream ss;
    ss << "Vec2Int { ";
    ss << "x: ";
    ss << x;
    ss << ", ";
    ss << "y: ";
    ss << y;
    ss << " }";
    return ss.str();
}

bool Vec2Int::operator ==(const Vec2Int& other) const {
    return x == other.x && y == other.y;
}

size_t std::hash<Vec2Int>::operator ()(const Vec2Int& value) const {
    size_t result = 0;
    result ^= std::hash<int>{}(value.x) + 0x9e3779b9 + (result << 6) + (result >> 2);
    result ^= std::hash<int>{}(value.y) + 0x9e3779b9 + (result << 6) + (result >> 2);
    return result;
}